### PR TITLE
rose edit: add rename section menu item

### LINKED ISSUE #907: rose edit: rename sections menu option
We should support renaming sections from within `rose edit`.

### PR DESCRIPTION
**Pull request by benfitzpatrick**
This closes #907.

@arjclark, please review.
@matthewrmshin, please review 2.

### CODE REVIEW

**Comment by arjclark** [comment]
Playing around with the demo app in /demo/rose-config-edit/test.

If I rename, undo_5 to undo_4 it complains and then appears to merge the two together. I don't think this should be allowed i.e. if I enter a name for the section that already exists the OK button should be disabled.

**Comment by arjclark** [comment]
Renaming undo_3 to undo_2 highlights why just merging is bad - ultimately you will lose one of the options.

**Comment by benfitzpatrick** [comment]
Renaming to an existing section now disallowed!

**Comment by arjclark** [comment]
Looks ok to me. Manually testing as working.

Over to you @oliver-sanders

**Comment by arjclark** [comment]
Oops, passed to wrong person.

Tested as working, over to @matthewrmshin

File: lib/python/rose/config_editor/nav_panel_menu.py [hunk 1]
```diff
@@ -270,6 +270,34 @@ def remove_request(self, base_ns):
...
+        config_names = self.data.config.keys()
+        if base_ns is not None and '/' in base_ns:
+            config_name, subsp = self.util.split_full_ns(self.data, base_ns)
+            prefer_name_sections = {
+                config_name:
+                self.data.helper.get_sections_from_namespace(base_ns)}
+        else:
+            prefer_name_sections = {}
+        config_sect_dict = {}
+        sorter = rose.config.sort_settings
```
Comment (matthewrmshin) [inline lib/python/rose/config_editor/nav_panel_menu.py]: `sorter` is not used here.

File: lib/python/rose/config_editor/nav_panel_menu.py [hunk 2]
```diff
@@ -270,6 +270,34 @@ def remove_request(self, base_ns):
             for stack_item in self.undo_stack[start_stack_index:]:
                 stack_item.group = group
 
+    def rename_dialog(self, base_ns):
+        """Handle a rename section dialog and request."""
+        config_names = self.data.config.keys()
```
Comment (matthewrmshin) [inline lib/python/rose/config_editor/nav_panel_menu.py]: `config_names` used only once below. Probably no need to assign it to a local variable.

**Comment by benfitzpatrick** [comment]
Feedback addressed.

**Comment by arjclark** [comment]
@matthewrmshin - I'm happy with the changes if you are.

**Pull request merged**